plugin/src/main/java/org/opensearch/ml/action/model_group/TransportRegisterModelGroupAction.java

```diff
@@ -77,9 +84,35 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegi
     public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<String> listener) {
         try {
             String modelName = input.getName();
+            User user = RestActionUtils.getUserContext(client);
             try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                MLModelGroupBuilder builder = MLModelGroup.builder();
+                if (Boolean.TRUE.equals(input.getIsPublic())) {
+                    builder = builder.access(MLModelGroup.PUBLIC);
+                } else if (Boolean.TRUE.equals(input.getAddAllBackendRoles())) {
```

Comment:
Can user choose not to add any backend roles which means it's a private model that can only be accessed by the owner self?